libcnb/src/exec_d.rs

```diff
@@ -0,0 +1,25 @@
+use libcnb_data::exec_d::ExecDProgramOutput;
+use std::fs::File;
+use std::io::BufWriter;
+use std::io::Write;
+
+/// Writes the output of a CNB exec.d program in a spec compliant way.
+pub fn write_exec_d_program_output<O: Into<ExecDProgramOutput>>(o: O) {
+    // Allow compilation of exec.d programs under windows, but fail at runtime:
+    #[cfg(target_family = "windows")]
+    unimplemented!("libcnb.rs does not support running in Windows containers yet!");
+
+    #[cfg(target_family = "unix")]
+    {
+        use std::os::unix::io::FromRawFd;
+
+        // https://github.com/buildpacks/spec/blob/main/buildpack.md#execd
+        let output_file = unsafe { File::from_raw_fd(3) };
+
+        let serialized_output =
+            toml::to_string(&o.into()).expect("Could not TOML serialize exec.d program output: ");
```

Comment:
I presume we're fine with the two `expect`s here, since it's not worth having anything more heavyweight for error handling, given (a) this is for exec.d binaries which we want as small/simple as possible, (b) both of these failure modes are unlikely internal error modes users should not hit?

Reply:
Also what would an exec.d program do when these errors occur? There is no possible recovery when we're not able to write the output and/or serialize the TOML. The only thing implementations could do is logging/panicing.

I could see a use-case where someone would want to log errors to a central service which would be impossible for these two errors. I don't think it's worth adding a `write_exec_d_program_output` variant that returns a `Result` at this point - we can easily add this later if we need to.